web/src/main/webapp/xslt/common/index-utils.xsl

```diff
@@ -153,6 +153,17 @@
       "title": "<xsl:value-of select="util:escapeForJson($recordTitle)"/>",
       "origin": "<xsl:value-of select="normalize-space($origin)"/>"
       }</recordLink>
+    <xsl:variable name="fieldName"
+                  select="concat('recordLink_', $type, string-join($otherProperties//p[@name != '']/concat(@name, @value), '_'))"/>
```

Comment:
This looks a very specific filtering field, indexing a different field based on the `associationType` and `initiativeType`. I'm fine with that, but I'm not really sure about the value for most users. 

Would not be good also to index similar fields, just considering the `$type` value?

Reply:
The `type` is in the field name so we will have fields like `recordLink_sources` also. Association and initiative types are ISO codelist so values are well known eg.  https://github.com/geonetwork/core-geonetwork/blob/main/schemas/iso19139/src/main/plugin/iso19139/loc/eng/codelists.xml#L374